ignore emails that are from the reply by email addresses

ryantm opened the pull request:
discussion here: https://meta.discourse.org/t/spam-with-from-address-set-to-reply-by-email-address/87786/2

ZogStriP commented on lib/email/receiver.rb:
```diff
@@ -107,6 +108,7 @@ def create_incoming_email
     def process_internal
       raise BouncedEmailError  if is_bounce?
       raise NoSenderDetectedError if @from_email.blank?
+      raise FromReplyByAddressError if is_from_reply_by_email_address?
```
What happens when that exception is raised?

ryantm replied:
I think it gets caught by https://github.com/discourse/discourse/blob/master/lib/email/processor.rb#L59 then handled by https://github.com/discourse/discourse/blob/master/lib/email/processor.rb#L71

ryantm replied:
Oh, I looked at that code more and it looks like it is probably still going to send a rejection mail in that case. I'll work on making it better.

ZogStriP commented on lib/email/receiver.rb:
```diff
@@ -203,6 +205,20 @@ def is_bounce?
     end
 
+    def is_from_reply_by_email_address?
+      reply_addresses = [SiteSetting.reply_by_email_address]
```
Why not use the `reply_by_email_address_regex` function?

ryantm replied:
reply_by_email_address_regex is too specific. I want to handle the case where the spammer has removed the + and the reply_key

ZogStriP replied:
Any chances you could merge both methods and add an argument?

ryantm commented:
Okay, I think I made it actually work now! And I think I addressed @ZogStriP's concerns about copying the regexp function: I changed the regexp to not match everything if it was empty, and I match, even if the "+code" part is missing.

ryantm commented:
ping @ZogStriP

ZogStriP commented:
Thanks 👍
